### PR TITLE
Implement routes events

### PR DESCRIPTION
**Pull request by YESENIA2012**
*Implemented route to create a new event

### CODE REVIEW

File: dbs/migrations/20230623205949-create-events-table.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      },
+      event_date: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+      event_time: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+      capacity_people_event: {
```
Comment (erick2014) [inline dbs/migrations/20230623205949-create-events-table.js]: lets `rename capacity_people_event` to `capacity`

File: src/controllers/eventController.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const Events = require("../models/eventsModel.js");
+
+class EventsController {
+  getAllEvents() {
+    const allEvents = Events.findAll({ raw: true });
+
+    return allEvents;
+  }
+
+  createEvent({ title, description, event_date, event_time, capacity }) {
```
Comment (erick2014) [inline src/controllers/eventController.js]: lets rename the method to `create`

File: src/controllers/eventController.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      description,
+      event_date,
+      event_time,
+      capacity,
+    });
+
+    return event;
+  }
+
+  async updateEvent(
```
Comment (erick2014) [inline src/controllers/eventController.js]: lets rename the method to `update`

File: src/controllers/eventController.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        capacity: capacity || Events.capacity,
+      },
+      { where: { id: eventId } }
+    );
+    const eventUpdate = await Events.findOne({ where: { id: eventId } });
+
+    return eventUpdate;
+  }
+
+  deleteEvent(eventId) {
```
Comment (erick2014) [inline src/controllers/eventController.js]: lets rename the method to `delete`

File: src/controllers/eventController.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      {
+        title: title || Events.title,
+        description: description || Events.description,
+        event_date: event_date || Events.event_date,
+        event_time: event_time || Events.event_time,
+        capacity: capacity || Events.capacity,
+      },
+      { where: { id: eventId } }
+    );
+    const eventUpdate = await Events.findOne({ where: { id: eventId } });
```
Comment (erick2014) [inline src/controllers/eventController.js]: remove this, we need to find a way to return it when calling `Events.update`

Comment (YESENIA2012) [reply src/controllers/eventController.js]: MariaDB, being a fork of MySQL, also lacks the returning option to return updated records after an update operation.

In the case of MariaDB, the alternatives are similar to those mentioned above for MySQL:

Perform a second query: After executing the update operation, you can perform a second query to get the updated records using the unique identifier (such as the ID) of the changed record.

Using the affectedRows property: As in MySQL, Sequelize provides the affectedRows property after an update operation in MariaDB. You can use this property to infer whether the update was successful, and then perform an additional query to get the updated record.

In summary, the same considerations and approaches mentioned for MySQL are applicable to MariaDB in terms of getting the updated records after an update operation.

Comment (erick2014) [reply src/controllers/eventController.js]: alright lets create an object on line 26, called `dataToUpdate`, and setup all the properties to be updated inside, then return that object, make sure to push the id property before returning the response, and remove this query.

File: src/routes/events.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    }
+  }
+);
+
+eventRouter.delete("/:id", async (req, res, next) => {
+  const eventId = parseInt(req.params.id);
+
+  try {
+    await eventsController.delete(eventId);
+    res.sendStatus(204);
```
Comment (erick2014) [inline src/routes/events.js]: res.send({success:true})

File: src/routes/events.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    const eventId = parseInt(req.params.id);
+
+    try {
+      const eventUpdated = await eventsController.update(
+        { title, description, event_date, event_time, capacity },
+        eventId
+      );
+
+      if (!eventUpdated) {
+        return res.status(404).json({ error: "Event not found" });
```
Comment (erick2014) [inline src/routes/events.js:49]: return res.status(400).json({ error: "Event not found" });

**Review by erick2014 — APPROVED** [review]
